syft/source/source.go

```diff
@@ -114,6 +114,7 @@ func getImageWithRetryStrategy(userInput, location string, imageSource image.Sou
 		err,
 	)
 
+	imageSource = image.DetermineImagePullSource(userInput)
```

Comment:
nit: a healthy comment as to why the image source is redetermined would go a long way